components/builder-web/app/client/builder-api.ts

```diff
@@ -20,6 +20,10 @@ import { AppStore } from '../app.store';
 import { addNotification, signOut } from '../actions/index';
 import { WARNING } from '../actions/notifications';
 
+
+export enum ErrorCode {
```

Comment:
Would it be worth having a separate error class for apis? Maybe not since there's only one.

Reply:
Enum should probably be fine for now, yeah. I'll also make a card for generating these based on our actual [protocols](https://github.com/habitat-sh/builder/tree/master/components/builder-protocol/protocols) via [protobuf.js](https://github.com/dcodeIO/protobuf.js), which we did on another project recently.